content/cpp/concepts/inheritance/inheritance.md

```diff
@@ -0,0 +1,50 @@
+---
+Title: 'Inheritance'
+Description: 'Inheritance is the ability to base a new class on an existing class, creating a class that starts out with the same properties and methods as the existing class.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Inheritance'
+  - 'Objects'
+  - 'OOP'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+Inheritance is the ability to base a new class on an existing class, creating a class that starts out with the same properties and methods as the existing class. The class that inherits these properties and methods is called the "derived class" or "sub-class". The class that the sub-class inherits from is called the "base class".
+
+Inheritance is generally used when it's necessary to implement a number of different objects that are all of the same type, sharing several characteristics in common.
+A good example is a `Shape` class. It might have a `center` property with x,y coordinates, a `weight` property that defines a line width, and a `color` property that would define a fill color. Derived from this base class might be a `Rectangle` which would have `height` and `width` properties, and a `Circle` that would have a `radius` property. Both `Circle` and `Rectangle` would share the properties and methods of their base class, `Shape`, as well as the additional properties and methods that are in their own class definition.
+
+## Syntax
```

Comment:
```suggestion

## Example

A good example is a `Shape` class that might include:

- A `center` property with (x, y) coordinates.
- A `weight` property that defines a line width.
- A `color` property that would define a fill color. 

Some possible sub-classes derived from this base class might be:

- A`Rectangle` class which would have `height` and `width` properties.
- A `Circle` class that would have a `radius` property.

Both the `Circle` and `Rectangle` classes would share the properties and methods of their base `Shape` class, `Shape`, in addition to properties and methods unique to their own class definition.
```
Could the sub-classes in this example, perhaps, also have the following:

- One shared method from the base `Shape` class (maybe one for finding the area?)
- At least one unique method for each sub-class (ex. Circumference for `Circle` and Perimeter for `Rectangle`)